Holovibes/sources/compute/analysis.cc

```diff
@@ -477,4 +490,52 @@ void Analysis::insert_otsu()
         });
 }
 
+void Analysis::insert_bwareafilt()
+{
+    LOG_FUNC();
+
+    fn_compute_vect_.conditional_push_back(
+        [=]()
+        {
+            if (setting<settings::ImageType>() == ImgType::Moments_0 && setting<settings::BwareafiltEnabled>() == true)
+            {
+                size_t size = buffers_.gpu_postprocess_frame_size;
+
+                float* image_d = buffers_.gpu_postprocess_frame.get();
+                size_t* labels_d = size_t_buffer_1_.get();
+                size_t* labels_sizes_d = size_t_buffer_2_.get();
+                size_t* linked_d = size_t_buffer_3_.get();
+
+                size_t nb_labels;
+                size_t n = setting<settings::BwareafiltN>();
+                size_t* labels_max_d = nullptr;
+
+                // shift_corners(image_d, 1, fd_.width, fd_.height, stream_);
+
+                get_connected_component(image_d,
+                                        labels_d,
+                                        labels_sizes_d,
+                                        linked_d,
+                                        size_t_gpu_.get(),
+                                        fd_.width,
+                                        fd_.height,
+                                        stream_);
+                nb_labels = get_nb_label(labels_sizes_d, size, size_t_gpu_.get(), stream_);
+
+                if (nb_labels < n)
+                    n = nb_labels;
+                if (n)
+                {
+                    cudaXMalloc(&labels_max_d, n * sizeof(size_t));
+                    get_n_max_index(labels_sizes_d, size, labels_max_d, n, stream_);
+
+                    create_is_keep_in_label_size(labels_sizes_d, size, labels_max_d, n, stream_);
+
+                    area_filter(image_d, labels_d, size, labels_sizes_d, stream_);
+                    cudaXFree(labels_max_d);
```

Comment:
Maybe you could malloc in the constructor or in an Env because here you Malloc and Free at each pipe cycle that could affect perfs